Add support for react-native-reanimated v4 peer dependency

### Description

  Currently, @shopify/react-native-skia@2.2.3 has a
  peer dependency on react-native-reanimated@^3.0,
  which prevents installation in projects using
  react-native-reanimated v4.

  **Current behavior:**
  When trying to install dependencies in a project with
   react-native-reanimated@4.0.1, npm fails with:

  npm error Could not resolve dependency:
  npm error peerOptional react-native-reanimated@"^3.0"
   from @shopify/react-native-skia@2.2.3

  **Expected behavior:**
  @shopify/react-native-skia should support
  react-native-reanimated v4, which is now stable and
  being adopted by many React Native projects.

### React Native Skia Version

2.2.3

### React Native Version

0.79.5

### Using New Architecture

- [x] Enabled

### Steps to Reproduce

1. Create a new React Native project (or use an
  existing one)
  2. Install react-native-reanimated v4 with its
  required dependency react-native-worklets:
  npm install react-native-reanimated@4.0.1
  react-native-worklets@0.4.1
  2. Note: react-native-worklets is a required
  dependency for react-native-reanimated v4
  3. Try to install @shopify/react-native-skia:
  npm install @shopify/react-native-skia@2.2.3
  4. npm will fail with the peer dependency error:
  npm error code ERESOLVE
  npm error ERESOLVE could not resolve
  npm error
  npm error While resolving:
  @shopify/react-native-skia@2.2.3
  npm error Found: react-native-reanimated@4.0.1
  npm error
  npm error Could not resolve dependency:
  npm error peerOptional react-native-reanimated@"^3.0"
   from @shopify/react-native-skia@2.2.3

  Minimal package.json to reproduce:
  {
    "dependencies": {
      "react": "19.0.0",
      "react-native": "0.79.5",
      "react-native-reanimated": "4.0.1",
      "react-native-worklets": "0.4.1",
      "@shopify/react-native-skia": "2.2.3"
    }
  }

### Snack, Code Example, Screenshot, or Link to Repository

  Minimal package.json to reproduce:
  {
    "dependencies": {
      "react": "19.0.0",
      "react-native": "0.79.5",
      "react-native-reanimated": "4.0.1",
      "react-native-worklets": "0.4.1",
      "@shopify/react-native-skia": "2.2.3"
    }
  }

--- a/packages/skia/package.json
+++ b/packages/skia/package.json
@@ -78,7 +78,7 @@
   "peerDependencies": {
     "react": ">=19.0",
     "react-native": ">=0.78",
-    "react-native-reanimated": "^3.0"
+    "react-native-reanimated": ">=3.0"
   },
   "peerDependenciesMeta": {
     "react-native": {

